legacy/web/src/main/java/org/jboss/as/web/WebMigrateOperation.java

```diff
@@ -115,6 +117,20 @@
     private static final OperationStepHandler DESCRIBE_MIGRATION_INSTANCE = new WebMigrateOperation(true);
     private static final OperationStepHandler MIGRATE_INSTANCE = new WebMigrateOperation(false);
     public static final PathElement DEFAULT_SERVER_PATH = pathElement(Constants.SERVER, "default-server");
+    public static final String MIGRATE = "migrate";
+    public static final String MIGRATION_WARNINGS = "migration-warnings";
+    public static final String MIGRATION_ERROR = "migration-error";
+    public static final String MIGRATION_OPERATIONS = "migration-operations";
+    public static final String DESCRIBE_MIGRATION = "describe-migration";
+
+
+    public static final StringListAttributeDefinition MIGRATION_WARNINGS_ATTR = new StringListAttributeDefinition.Builder(MIGRATION_WARNINGS)
+            .setAllowNull(true)
+            .build();
+
+    public static final StringListAttributeDefinition MIGRATION_ERROR_ATTR = new StringListAttributeDefinition.Builder(MIGRATION_ERROR)
+            .setAllowNull(true)
+            .build();
```

Comment:
This one can either be:

SimpleAttributeDefinition, ModelType.OBJECT

or, if you are motivated to describe "operation" and "result", then you can do that and use ObjectTypeAttributeDefinition.Builder. But you can't fully describe "operation" and "result" (you don't know the params of "operation" and the type of result.failure-description, so IMO going with SimpleAttributeDefinition, ModelType.OBJECT is fine.